src/django/api/admin.py

```diff
@@ -270,18 +274,91 @@ class Meta:
             'source_by',
             'base_url',
             'display_text',
-            'json_schema'
+            'json_schema',
+            'active',
+            'system_field'
         ]
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
+    def clean(self):
+        '''
+        Validate that protected fields of system fields are not modified.
+        '''
+        cleaned_data = super().clean()
+
+        # Only validate for existing system fields.
+        if self.instance and self.instance.pk and self.instance.system_field:
+            from api.models.partner_field import PartnerField
+
+            try:
+                # Get the original instance from database.
+                original = PartnerField.objects \
+                    .get_all_including_inactive() \
+                    .get(pk=self.instance.pk)
+
+                protected_fields = {
+                    'name': 'Name',
+                    'type': 'Type',
+                    'json_schema': 'JSON Schema',
+                    'system_field': 'System Field'
+                }
+
+                for field_name, field_label in protected_fields.items():
+                    original_value = getattr(original, field_name)
+                    current_value = cleaned_data.get(field_name)
+
+                    if original_value != current_value:
+                        # Add field-specific error.
```

Comment:
According to best practices the comments should answer the question `Why` something is being done a certain way, this seems like it's just describing `What` is being done.